### ISSUE TITLE
Helm v3: lint silently ignores non existing packaged charts

### ISSUE THREAD

**Issue by andytom** [helm/helm#6428]
Output of `helm version`:
```
$ helm3 version
version.BuildInfo{Version:"v3.0.0-beta.3", GitCommit:"5cb923eecbe80d1ad76399aee234717c11931d9a", GitTreeState:"clean", GoVersion:"go1.12.9"}

$ helm version --client
Client: &version.Version{SemVer:"v2.14.3", GitCommit:"0e7f3b6637f7af8fcfddb3d2941fcc7cbebb0085", GitTreeState:"clean"}
```
Output of `kubectl version`:
N/A

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 
N/A

----

If you try to lint a packaged chart that doesn't exist, helm3 silently ignores it.
 
```
$ helm3 lint aaa-1.0.0.tgz
==> Linting aaa-1.0.0.tgz

1 chart(s) linted, 0 chart(s) failed
```

```
$ helm lint aaa-1.0.0.tgz
==> Skipping aaa-1.0.0.tgz
open aaa-1.0.0.tgz: no such file or directory

0 chart(s) linted, no failures
```

**Comment by karuppiah7890** [comment]
For helm2, looks like it was expected 🤔 as I can see a `Skipping` message. Let me check code. So, what's expectation? Both helm2 and helm3 should say that there was one failure and set an exit code? Or helm3 should work similar to helm2's current flow ?

**Comment by karuppiah7890** [comment]
Also, if something was skipped, I think it would be better to show
```
$ helm lint aaa-1.0.0.tgz
==> Skipping aaa-1.0.0.tgz
open aaa-1.0.0.tgz: no such file or directory

0 chart(s) linted, 1 chart(s) skipped, no failures
```

**Comment by andytom** [comment]
> For helm2, looks like it was expected 🤔 as I can see a `Skipping` message. Let me check code. So, what's expectation? Both helm2 and helm3 should say that there was one failure and set an exit code? Or helm3 should work similar to helm2's current flow ?

I think the helm2 behaviour is fine. For helm3 I'm not sure what we should do, if we display an error and set a non-zero exit code if the file doesn't exist I don't think that would be too surprising on the other hand following the same flow as helm2 might make the upgrade easier.

**Comment by waveywaves** [comment]
@karuppiah7890 We should have an error message which notifies the user about the non existence of file and write to stderr with an nonzero exit code as @andytom said. Here we haven't really skipped because there was nothing to skip. Hence erroring out makes more sense. Skipping would mean that the chart exists but there was a problem in parsing hence it was "skipped".

**Comment by karuppiah7890** [comment]
I would think the same. Seeing exit code is also helpful in pipelines. I would like to see if anyone has different expectations. Also @andytom also mentioned that upgrading would be easier if behavior is same as helm 2. Not sure how to take a decision here. I'll check more about how it behaves in both first

**Comment by karuppiah7890** [comment]
@hickeyma any thoughts? 😄

**Comment by hickeyma** [comment]
I think it should probably work as it does in Helm 2. My reasoning is that it is asked to lint a package/archive that doesn't exist. It states it doesn't exist. It didn't actually perform a lint, hence no results, chart skipped.

Open to other opinions though.

**Comment by hickeyma** [comment]
In any case, would someone like to do the PR?

**Comment by karuppiah7890** [comment]
I would like to do a PR on this. I just want to get the expectation and the approach straight. I'm guessing the helm2 approach was taken for the above reason. I wouldn't change unless there any decision somewhere that this needs thoughts and change. I did like the set exit code approach - just in case someone makes a typo and lint is running in some CI pipeline (which I have never done though) then exit codes help and some people have nice prompts to show exit code status too. Just a nice user experience that something went wrong and not exactly as expected. Then the message on the screen could say "xyz chart missing" as one of the logs

**Comment by hickeyma** [comment]
@karuppiah7890 Go for it.

**Comment by karuppiah7890** [comment]
@hickeyma which approach are we going with? helm2 way?

**Comment by andytom** [comment]
Personally I think it would make more sense to return an error and set the exit code if the tarball doesn't exist, as @karuppiah7890 has pointed out it would make it obvious if someone made a typo when running it locally or cause faulty CI pipelines to fail. This would also be consistent with trying to lint a non existing directory.

```
helm3 lint /no/such/chart  aaa-1.0.0.tgz
==> Linting /no/such/chart
Error no chart found for linting (missing Chart.yaml)

==> Linting aaa-1.0.0.tgz

Error: 2 chart(s) linted, 1 chart(s) failed
```

**Comment by hickeyma** [comment]
Sounds good. @karuppiah7890 Can you put the new approach in the description of the PR to notify the reviewers?

**Comment by karuppiah7890** [comment]
Sure! Will do that!

**Comment by karuppiah7890** [comment]
Just mentioning a weird thing I noticed just now. For some non existent `.tgz` alone helm lint does not fail , but for others it fails with proper exit code! I will try to fix this too!

```
$ helm lint blah.tgz
==> Skipping blah.tgz
open blah.tgz: no such file or directory

0 chart(s) linted, no failures

$ helm lint blah.tar.gz
==> Skipping blah.tar.gz
No chart found for linting (missing Chart.yaml)

Error: 0 chart(s) linted, 1 chart(s) failed

$ helm lint blah
==> Skipping blah
No chart found for linting (missing Chart.yaml)

Error: 0 chart(s) linted, 1 chart(s) failed
```

**Comment by karuppiah7890** [comment]
This can be closed as #6536 has been merged
